app/views/foreman/hosts/_registration_commands.html.erb

```diff
@@ -1,12 +1,12 @@
 <div class='form-group'>
   <label class='col-md-2 control-label'>
-    <%= _('Activation Key(s)') %>
-    <% help = _('Activation key(s) for Subscription Manager. Required for CentOS and Red Hat Enterprise Linux. Multiple keys add separated by comma, example: key1,key2,key3.') %>
+    <%= _('Activation Key(s) *') %>
+    <% help = _('Activation key(s) for Subscription Manager. Multiple keys add separated by comma, example: key1,key2,key3.') %>
     <a rel="popover" data-content="<%= help %>" data-trigger="focus" data-container="body" data-html="true" tabindex="-1">
       <span class="pficon pficon-info "></span>
     </a>
   </label>
   <div class='col-md-4'>
-    <%= text_field_tag 'activation_key', params[:activation_key], class: 'form-control' %>
+    <%= text_field_tag 'activation_key', params[:activation_key], class: 'form-control', required: true %>
```

Comment:
This will make the field required regardless of which OS is selected.  Is that the intention?

Reply:
Yes, for Foreman with Katello we want to make it required regardless of which OS is selected.
This will be changed in Foreman 2.5, where Activation keys will be hidden as advanced field,
and users will use host group as primary source of the data.
There is a redmine PR for it: https://projects.theforeman.org/issues/31240